### ISSUE TITLE
Player selection box is wrong dimensions

### ISSUE THREAD

**Issue by GreenXenith** [minetest/minetest#6152]
I don't know what commit did this or anything, and I don't know if someone already reported this, but I noticed it.
![screenshot_20170718_104653](https://user-images.githubusercontent.com/24834740/28331631-a8027756-6ba6-11e7-8956-44653c3e7e2f.png)
As you can see, the selection box is too thin and too short. The head sticks out the top a tiny bit and the arms stick out the sides. If this was intentional, it looks horrible..

stujones11 pointed out
> Currently the player collision box is hard-coded in the engine. AFAIK the last commit to change this was minetest/minetest@2d5bd7f

**Comment by kaadmy** [comment]
Wouldn't the actual solution be to allow mods to change the player's selection box, and possibly the collision box?

**Comment by stujones11** [comment]
Entity selection and collision box are the same thing, IIRC. There is a PR to make the player collision box settable #6007

**Comment by paramat** [comment]
3 commits affected collisionbox b6f4a9c7e1a4f0bac66fd6f6ff844425ac775975 2d5bd7f414f8b8107254490af2360d6e29f1a8d2 da88a186766020762f5c86bc6ba1624e4feaae2c
The collisionbox has always been 1.75 nodes tall:
```
 bool PlayerSAO::getCollisionBox(aabb3f *toset) const
 {
	*toset = aabb3f(-BS * 0.30, 0.0, -BS * 0.30, BS * 0.30, BS * 1.75, BS * 0.30);
```
But my change set that box as a default in `PlayerSAO::PlayerSAO(ServerEnvironment *env_, RemotePlayer *player_, u16 peer_id_, bool is_singleplayer):`
```
-	m_prop.collisionbox = aabb3f(-1/3.,-1.0,-1/3., 1/3.,1.0,1/3.);
+	m_prop.collisionbox = aabb3f(-0.3f, -1.0f, -0.3f, 0.3f, 0.75f, 0.3f);
```
This change was not reverted because it is correct for the player, however if the head sticks out i can raise it a little.
If the selectionbox is set to the collisionbox the width is intentional and can't be changed.

**Comment by paramat** [comment]
![screenshot_20170719_145409](https://user-images.githubusercontent.com/3686677/28370455-3df18202-6c92-11e7-9bba-e4c4217eb931.png)

Confirmed selectionbox is set by `m_prop.collisionbox = aabb3f(-0.3f, -1.0f, -0.3f, 0.3f, 0.75f, 0.3f);`

**Comment by paramat** [comment]
PR #6154

**Comment by stujones11** [comment]
I have taken a closer look in game and noticed that the hight issues is caused by the player levitating slightly above the ground. The height of 1.75 seem correct given that the player measures 28 pixels* compared next to a node 2 / 32 * 28 = 1.75

*AFAIK from the model, it should really be 32 pixels high so there is some discrepancy in pixel size there but that's likely a known issue.

**Comment by paramat** [comment]
Also, the player pixels are stretched slightly vertically so they are non-square, which bothers me, there's an issue about it here https://github.com/minetest/minetest_game/issues/1411 if anyone would like to edit the model (and at the same time make the head top at exactly 1.75 above ground).
I assume MC players have square pixels, they look better proportioned to me, MT players are too elongated.

**Comment by GreenXenith** [comment]
@paramat what do you mean by "new"? Just resizing stuff or completely new stuff?

**Comment by stujones11** [comment]
@paramat to have perfectly square pixels that are equal in size to those of nodes the player model would be exactly two nodes high. This is a limitation of the skin format, not the model itself.

**Comment by paramat** [comment]
stujones11 yes i know, i mean the pixels on the player should be square, but of course should remain scaled smaller to keep player height the same.
By 'new' i mean my request in the MTG issue, square pixels for the body and limbs, just as the head has, but with same player height.
The MTG issue explains everything.

**Comment by stujones11** [comment]
> The MTG issue explains everything.

I have replied there :)

**Comment by paramat** [comment]
Fixed in #6007
